### ISSUE TITLE
cmake error /usr/bin/ld: cannot find -lpthreads

### ISSUE THREAD

**Issue by richardweib** [BVLC/caffe#5563]
### Issue summary

I'm use cmake to compile the caffe with only_cpu,but I get the error like this:

```
/usr/bin/cmake -E cmake_progress_report /opt/caffe/build/CMakeFiles/CMakeTmp/CMakeFiles 1
Building C object CMakeFiles/cmTryCompileExec1767991674.dir/CheckFunctionExists.c.o
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create   -o CMakeFiles/cmTryCompileExec1767991674.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-2.8/Modules/CheckFunctionExists.c
Linking C executable cmTryCompileExec1767991674
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTryCompileExec1767991674.dir/link.txt --verbose=1
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create    CMakeFiles/cmTryCompileExec1767991674.dir/CheckFunctionExists.c.o  -o cmTryCompileExec1767991674 -rdynamic -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
``` 

### Steps to reproduce

1.install the dependencies
2.git clone the project
3. pip install the requirements
4.mkdir build && cd build && cmake -DCPU_ONLY=1 

### Your system configuration
Operating system: Ubuntu 14.04
Compiler: gcc-4.9

**Comment by knhemanth** [comment]
I have the same issue. I think "-lpthreads" is a typo. It should probably be "-lpthread" according to [this](http://stackoverflow.com/questions/37967086/usr-bin-ld-cannot-find-lpthreads). Seems like a cmake issue to me but I am not sure.

**Comment by amanjhunjhunwala** [comment]
Same issue affects me...Any solution to this?

**Comment by deepali-c** [comment]
Yes, this error seen in `CMakeError.log` is due to the way `cmake` uses to look for `pthread_create `method. 

So if `cmake ..`shows the following:
```
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
```
Then all is well.

**Comment by w5688414** [comment]
I met the same error, can anyone help me?

**Comment by w5688414** [comment]
my error message is below:
`Run Build Command:"/usr/bin/make" "cmTC_2f64b/fast"
/usr/bin/make -f CMakeFiles/cmTC_2f64b.dir/build.make CMakeFiles/cmTC_2f64b.dir/build
make[1]: Entering directory '/home/idc/TextBoxes/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_2f64b.dir/CheckFunctionExists.c.o
/usr/bin/cc    -DCHECK_FUNCTION_EXISTS=pthread_create   -o CMakeFiles/cmTC_2f64b.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.5/Modules/CheckFunctionExists.c
Linking C executable cmTC_2f64b
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_2f64b.dir/link.txt --verbose=1
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create    CMakeFiles/cmTC_2f64b.dir/CheckFunctionExists.c.o  -o cmTC_2f64b -rdynamic -lpthreads 
/usr/bin/ld: 找不到 -lpthreads
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_2f64b.dir/build.make:97: recipe for target 'cmTC_2f64b' failed
make[1]: *** [cmTC_2f64b] Error 1
make[1]: Leaving directory '/home/idc/TextBoxes/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_2f64b/fast' failed
make: *** [cmTC_2f64b/fast] Error 2
`

**Comment by deepali-c** [comment]
Yes, even I get that error, because `cmake `first checks for `pthread_create` in `pthreads `and then it checks it in `pthread` . 

Could you please post complete output of `cmake ..`

**Comment by fvilca** [comment]
someone?
I met the same error, can anyone help me?

**Comment by gldneagl** [comment]
Anyone willing to supply this answer?

I have the same errors showing up during build after running make command

-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes

**Comment by mjjdick** [comment]
I met the same error, does anyone solve this problem?

**Comment by dean09131** [comment]
I also met this problem, my OS is Ubuntu 16.04, and I really have the pthread.so, but it won't use it. On the Internet, most people say it is should use "gcc -pthread ....", seems that the code have the "gcc -lpthreads...", but I not sure, and hope the developer could help us.
